File: app/src/main/java/com/mmjang/ankihelper/data/dict/MaldpeDatabase.java
package com.mmjang.ankihelper.data.dict;

import android.content.Context;

import androidx.room.Dao;
import androidx.room.Database;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.Query;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.mmjang.ankihelper.data.dict.Form; // Import the Form class

@Database(entities = {DummyEntity.class}, version = 1, exportSchema = false)
public abstract class MaldpeDatabase extends RoomDatabase {

    public abstract MaldpeDao maldpeDao();

    private static MaldpeDatabase instance;

    public static synchronized MaldpeDatabase getInstance(Context context) {
        if (instance == null) {
            instance = Room.databaseBuilder(context.getApplicationContext(),
                    MaldpeDatabase.class, "maldpe.db")
                    // Removed .allowMainThreadQueries() - use coroutines for async operations
                    .build();
        }
        return instance;
    }
}

File: app/src/main/java/com/lmyby/ankiquicker/data/dict/MaldpeDatabase.kt
package com.lmyby.ankiquicker.data.dict

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

/**
 * Converted to Kotlin as part of Phase 2 data model migration
 */
@Database(entities = [DummyEntity::class], version = 2, exportSchema = false)
abstract class MaldpeDatabase : RoomDatabase() {

    abstract fun maldpeDao(): MaldpeDao

    companion object {
        @Volatile
        private var instance: MaldpeDatabase? = null

        @JvmStatic
        fun getInstance(context: Context): MaldpeDatabase {
            return instance ?: synchronized(this) {
                instance ?: Room.databaseBuilder(
                    context.applicationContext,
                    MaldpeDatabase::class.java,
                    "maldpe.db"
                )
                    .createFromAsset("databases/maldpe.db")
                    .fallbackToDestructiveMigration()
                    // Removed .allowMainThreadQueries() - use coroutines for async operations
                    .build().also { instance = it }
            }
        }
    }
}
